aof/rewrite.go

```diff
@@ -117,9 +96,7 @@ func (persister *Persister) FinishRewrite(ctx *RewriteCtx) {
 		logger.Error("open aofFilename failed: " + err.Error())
 		return
 	}
-	defer func() {
-		_ = src.Close()
-	}()
+
 	_, err = src.Seek(ctx.fileSize, 0)
```

Comment:
可以试试通过一个匿名函数的 defer 来确保关闭了 src 句柄。
注意，匿名函数里的 err 是从外部捕获的， 这样才可以在匿名函数退出后通过 if err != nil 判断中间是否遇到了错误

```suggestion
        tmpFile := ctx.tmpFile
	src, err := os.Open(persister.aofFilename)
	if err != nil {
		logger.Error("open aofFilename failed: " + err.Error())
		return
	}
	
	func() {
		defer func() {
			_ = src.Close()
		}()
		// write commands executed during rewriting to tmp file
		_, err = src.Seek(ctx.fileSize, 0)
		if err != nil {
			err = errors.New("seek failed: " + err.Error())
			return
		}

		// sync tmpFile's db index with online aofFile
		data := protocol.MakeMultiBulkReply(utils.ToCmdLine("SELECT", strconv.Itoa(ctx.dbIdx))).ToBytes()
		_, err = tmpFile.Write(data)
		if err != nil {
			err = errors.New("tmp file rewrite failed: " + err.Error())
			return
		}
		// copy data
		_, err = io.Copy(tmpFile, src)
		if err != nil {
			err = errors.New("copy aof filed failed: " + err.Error())
			return
		}
	}()
	if err != nil {
		logger.Error(err)
		return
	}

	tmpFileName := tmpFile.Name()
	_ = tmpFile.Close()
```